GoogleForCreators/web-stories-wp
# Workspace: Font Dropdown with Typeface Previews

## Summary

- Add font previews to font dropdown.

## Relevant Technical Choices

This was originally implemented in this PR, but in a too complex way with a single huge component.

I restructured the current code a bit and removed all the search and recent fonts stuff to be replaced by something much simpler in future PR's for those specific purposes.

## To-do

- [x] Add scrollable list knowing which items are visible
- [x] Preload visible font items using GFonts menu subset
- [x] Highlight currently selection font

## To-do in future PR's
- [ ] Show fonts used in this story at the top (~no issue, I think?~ #1955 )
- [ ] Add search in three sections  (#30)

## User-facing changes

Font previews are now visible in font selector.

## Testing Instructions

1. Open font drop down.
1. See previews of all fonts
1. See a checkmark next to the currently used font

---



Fixes #309
